File: agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/CompletionItemInfo.kt
@file:Suppress("FunctionName", "ClassName", "unused", "EnumEntryName", "UnusedImport")
package com.sourcegraph.cody.agent.protocol_generated;

import com.google.gson.annotations.SerializedName;

data class CompletionItemInfo(
  val parseErrorCount: Long? = null,
  val lineTruncatedCount: Long? = null,
  val truncatedWith: TruncatedWithEnum? = null, // Oneof: tree-sitter, indentation
  val nodeTypes: NodeTypesParams? = null,
  val nodeTypesWithCompletion: NodeTypesWithCompletionParams? = null,
  val lineCount: Long,
  val charCount: Long,
  val insertText: String? = null,
  val stopReason: String? = null,
) {

  enum class TruncatedWithEnum {
    @SerializedName("tree-sitter") `Tree-sitter`,
    @SerializedName("indentation") Indentation,
  }
}

File: agent/bindings/csharp/CompletionItemInfo.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class CompletionItemInfo
  {
    [JsonProperty(PropertyName = "parseErrorCount")]
    public int ParseErrorCount { get; set; }
    [JsonProperty(PropertyName = "lineTruncatedCount")]
    public int LineTruncatedCount { get; set; }
    [JsonProperty(PropertyName = "truncatedWith")]
    public string TruncatedWith { get; set; } // Oneof: tree-sitter, indentation
    [JsonProperty(PropertyName = "nodeTypes")]
    public NodeTypesParams NodeTypes { get; set; }
    [JsonProperty(PropertyName = "nodeTypesWithCompletion")]
    public NodeTypesWithCompletionParams NodeTypesWithCompletion { get; set; }
    [JsonProperty(PropertyName = "lineCount")]
    public int LineCount { get; set; }
    [JsonProperty(PropertyName = "charCount")]
    public int CharCount { get; set; }
    [JsonProperty(PropertyName = "insertText")]
    public string InsertText { get; set; }
    [JsonProperty(PropertyName = "stopReason")]
    public string StopReason { get; set; }
  }
}
